Updates for Singularity documentation - addressing bind mount potential issues. (#1237)
rcgsheffield/sheffield_hpc: bessemer/software/apps/singularity.rst

the paths ``/fastdata``, ``/home``, ``/scratch``, ``/shared`` are
automatically mounted to your Bessemer directories.

.. warning::

>   The automatic bind mounting of the ``/home`` path can result in shell environment corruption and unintended sharing of executables 

Reviewer: ```suggestion
  The automatic bind mounting of your HPC home directory into Singularity containers can result in the unexpected sharing of things like executables 
```

  and libraries between the host and Singularity container.

>   Unintended behaviour / corrupted Singularity environments may occur due to the presence of:

Reviewer: ```suggestion
  Unintended behaviour may occur with Singularity on the HPC system due to the presence of:
```

  * Shell initialisation files e.g. ``~/.bashrc`` or ``~/.profile``

>   * R profile files or libraries e.g. ``~/.Rprofile``

Reviewer: ```suggestion
  * R profile files (e.g. ``~/.Rprofile``) and/or libraries (e.g ``~/R/x86_64-pc-linux-gnu-library/4.1``)
```
